Add Team documentation (#713)
grafana/k6-docs: src/data/markdown/docs/03 cloud/05 Project and Team Management/02 Teams.md

---

title: 'Teams'
excerpt: 'Guide on setting up Teams in k6 Cloud'
---

> ⭐️ Teams are available on Enterprise plans.

> ## What are Teams?

Reviewer: ```suggestion
## A level between organizations and projects
```

> A _Team_ is an optional organizational level sitting between [Organizations](/cloud/project-and-team-management/organizations) and [Projects](/cloud/project-and-team-management/projects). With Teams, you can group the Projects of your Organizations into categories. All Team members gain access to the Projects under that Team. The Team can provide varying [Membership](/cloud/project-and-team-management/members) levels to each Project.

Reviewer: ```suggestion
A _Team_ is an optional organizational level that sits between [Organizations](/cloud/project-and-team-management/organizations) and [Projects](/cloud/project-and-team-management/projects).
With Teams, you can group the Projects of your Organizations into categories.
All Team members gain access to the Projects under that Team. The Team can provide varying [Membership](/cloud/project-and-team-management/members) levels to each Project.
```

> Using Teams can help when administrating k6 Organizations with a large number of members and projects. For example, you could have a "Backend" Team that grants read-write access to all backend-related k6-cloud Projects and a "Payment" Team dealing only with Projects testing payment-related services. The Organization-admin can add/remove a member from the Team rather than having to do so for every related Project. 

Reviewer: ```suggestion
Teams help you administrate k6 Organizations with a large number of members and projects.
For example, a "Backend" Team could grant read-write access to all backend-related k6-cloud Projects,
and a "Payment" Team might deal only with Projects for testing payment-related services.
In this case, an Organization admin could add or remove members from these Teams (rather than doing so for each project). 
```

> Team-membership can be mixed with regular, "direct", Project-membership. If access-level differs, you'll use the higher of the two. If you leave the Team, your regular Project-membership will not be affected. 

Reviewer: ```suggestion
You can mix Team membership with Project membership.
If the access-level differs, k6 uses the higher membership.
If you leave the Team, your regular Project membership does not change. 
```

> > Note that when you want to run k6 scrips in the cloud, it's still the [ProjectID](/cloud/project-and-team-management/projects#running-cli-tests-in-a-specific-project) of the Project you need to specify. The Team granting that project membership is never referenced in k6 scripts.

Reviewer: ```suggestion
<Blockquote mod="note" title="You still must specify the projectID in cloud scripts.">

That team that the project belongs to is never referenced in k6 scripts.
Refer to [Projects](/cloud/project-and-team-management/projects) for examples.

</Blockquote>

```

> ## Managing teams

Reviewer: ```suggestion
## Necessary organization structure and permissions
```

> To add Teams and Team members via the app, your Organization must first have some [Projects](/cloud/project-and-team-management/projects) and some [Organization Members](/cloud/project-and-team-management). You need **Organization admin** or **Organization read/write** privileges to add new Team. 

Reviewer: ```suggestion
To add Teams and Team members, you need the following:
- An Organization with [Projects](/cloud/project-and-team-management/projects) and some [Organization Members](/cloud/project-and-team-management).
- **Organization admin** or **Organization read/write** privileges 
```

> Using the [k6 cloud browser app](https://app.k6.io), you should be able to access _Teams_ view from the left side bar.

Reviewer: ```suggestion
## Managing teams

To manage teams from k6 Cloud, use the left sidebar and select **Teams**. 
```

![Team menu](images/05-Teams/teams-menu.png)

### Creating a team

> To create a new team click on **Create new team** button. A Team must have an organization-unique **name** and at least one selected **project**. You can also add an optional **description**.

Reviewer: ```suggestion
To create a new team:
1. Select the **Create new team** button.
2. Give the team an organization-unique **name**, and assign at least one **project**. You can also add an optional **description**.
```

![Creating a new Team](images/05-Teams/create-team.png)

Selected **projects** can provide different accesses:

> - **Read/Write** access means Team members will be able to look at, create and edit test-runs in the Project. They will not be able to delete or rename the Project itself.

Reviewer: ```suggestion
- **Read/Write** access allows Team members to view, create, and edit test runs in the Project. They cannot delete or rename the Project itself.
```

- **Read-only** access allows Team members to look at test data but not modify or create new test-runs.

### Inviting a team member

> Once the Team has been created, select it in your list of Teams and press **Invite members**. Only existing members of your Organization can be invited. When adding one or more member, consider what Team-level access they should have. Every batch of members you invite will join with the same Team-level role.

Reviewer: ```suggestion
Once the Team has been created, select it in your list of Teams and press **Invite members**.
You can invite only existing members of your Organization.
When adding one or more member, consider what Team-level access they should have. Every batch of members you invite will join with the same Team-level role.
```

![Inviting a new team member](images/05-Teams/invite-team-member.png)

> - All Team members gain access to all Projects linked to that Team (according to the Project accesses you set up for the Team)
> - **Team Admins** can modify the Team itself (change name and description) as well as invite/remove members from the Team. 

Reviewer: ```suggestion
- All Team members gain access to all Projects linked to that Team (according to the Project accesses you set up for the Team)
- **Team Admins** can modify the Team itself (change name and description) and also invite or remove members from the Team. 
```

> - **Team members** are read-only and can't modify the Team itself. They only makes use of the Team to access Projects.

Reviewer: ```suggestion
- **Team members** are read-only and can't modify the Team itself. They can use the Team only to access its Projects.
```

> > Don't confuse the _Team_ access level with the _Project_ access level granted by the Team!
>  Most members just need **read only** Team access.

Reviewer: ```suggestion
<Blockquote mod="attention" title="">

Don't confuse the _Team_ access level with the _Project_ access level granted by the Team!
 Most members just need **read only** Team access.

</Blockquote>

```

### Managing team members

> Once members are added, they appear in the list and Organization admins and Team admins for that Team will be able to add/remove members from it.

Reviewer: ```suggestion
Once members are added, they appear in the list.
Organization admins and Team admins can add or remove members from it.
```

---

title: 'Teams'
excerpt: 'Guide on setting up Teams in k6 Cloud'
---

> ⭐️ Teams are available on Enterprise plans.

## A level between organizations and projects

> A _Team_ is an optional organizational level that sits between [Organizations](/cloud/project-and-team-management/organizations) and [Projects](/cloud/project-and-team-management/projects). 

Reviewer: ```suggestion
A _Team_ is an optional structural unit between [Organizations](/cloud/project-and-team-management/organizations) and [Projects](/cloud/project-and-team-management/projects). 
```

With Teams, you can group the Projects of your Organizations into categories. 

> All Team members gain access to the Projects under that Team. The Team can provide varying [Membership](/cloud/project-and-team-management/members) levels to each Project.

Reviewer: ```suggestion
All users who belong to a team gain access to the Projects under that Team. The Team can provide varying [Membership](/cloud/project-and-team-management/members) levels to each Project.
```

> Teams help you administrate k6 Organizations with a large number of members and projects.

Reviewer: ```suggestion
Teams help you administrate k6 Organizations with a large number of users and projects.
```

For example, a "Backend" Team could grant read-write access to all backend-related k6-cloud Projects,
and a "Payment" Team might deal only with Projects for testing payment-related services.

> In this case, an Organization admin could add or remove members from these Teams (rather than doing so for each project).

Reviewer: ```suggestion
In this case, an Organization admin could add or remove users from these Teams (rather than doing so for each project).
```

- **Read-only** access allows Team members to look at test data but not modify or create new test-runs.

### Inviting a team member

Once the Team has been created, select it in your list of Teams and press **Invite members**.
You can invite only existing members of your Organization.
When adding one or more member, consider what Team-level access they should have. Every batch of members you invite will join with the same Team-level role.

![Inviting a new team member](images/05-Teams/invite-team-member.png)

- All Team members gain access to all Projects linked to that Team (according to the Project accesses you set up for the Team)
- **Team admins** can modify the Team itself (change name and description) and also invite or remove members from the Team.
- **Team members** are read-only and can't modify the Team itself. They can use the Team only to access its Projects.

<Blockquote mod="attention" title="">

Don't confuse the _Team_ access level with the _Project_ access level granted by the Team!
Most members just need **read only** Team access.

</Blockquote>

### Managing team members

Once members are added, they appear in the list.

> Organization admins and Team admins can add or remove members from it.

Reviewer: ```suggestion
Organization admins and Team admins can add or remove users from a team.
```
